### ISSUE TITLE
Minor Features and Security

### ISSUE THREAD

**Issue by kea137** [kea137/Bible#168]
Fix activity log page, when in this page navigating to other routes fail. Fix the caching in the bible page, it fails everytime when clicked. in verse link canvas, when verse is added or removed do not show modal for success instead show a sonner instead at the bottom righthand corner. in fact, make this kind of response for every other operation that throws a success or error message via modal. compile features documentation in one file, i.e unified search md and lesson feature documentation files should be removed and its content should be in one features documentation. create security analysis, assess vulnerabilities and fix them.
